diff -uNr  lib/5.10.0/os2/Config.pm.org lib/5.10.0/os2/Config.pm
--- lib/5.10.0/os2/Config.pm.org	2007-12-25 22:53:38.000000000 +0900
+++ lib/5.10.0/os2/Config.pm	2011-11-17 17:58:48.000000000 +0900
@@ -78,7 +78,7 @@
 
 # tie returns the object, so the value returned to require will be true.
 tie %Config, 'Config', {
-    archlibexp => '/perl5/lib/5.10.0/os2',
+    archlibexp => 'f:/lang/perl5/lib/5.10.0/os2',
     archname => 'os2',
     cc => 'gcc',
     d_readlink => 'define',
@@ -89,14 +89,14 @@
     inc_version_list => ' ',
     intsize => '4',
     ldlibpthname => '',
-    libpth => 'U:/usr/lib U:/usr/lib/gcc-lib/i386-pc-os2-emx/3.3.5 u:/extras/lib u:/dev/qt/trunk/lib u:/openldap/lib u:/cups/lib',
+    libpth => 'f:/usr/lib f:/usr/dll',
     osname => 'os2',
     osvers => '2.45',
     path_sep => ';',
-    privlibexp => '/perl5/lib/5.10.0',
-    scriptdir => '/perl5/bin',
-    sitearchexp => '/perl5/lib/site_perl/5.10.0/os2',
-    sitelibexp => '/perl5/lib/site_perl/5.10.0',
+    privlibexp => 'f:/lang/perl5/lib/5.10.0',
+    scriptdir => 'f:/lang/perl5/bin',
+    sitearchexp => 'f:/lang/perl5/lib/site_perl/5.10.0/os2',
+    sitelibexp => 'f:/lang/perl5/lib/site_perl/5.10.0',
     useithreads => undef,
     usevendorprefix => undef,
     version => '5.10.0',
diff -uNr  lib/5.10.0/os2/Config_heavy.pl.org lib/5.10.0/os2/Config_heavy.pl
--- lib/5.10.0/os2/Config_heavy.pl.org	2007-12-27 17:21:30.000000000 +0900
+++ lib/5.10.0/os2/Config_heavy.pl	2011-11-17 18:15:00.000000000 +0900
@@ -101,8 +101,8 @@
 api_version='10'
 api_versionstring='5.10.0'
 ar='ar'
-archlib='/perl5/lib/5.10.0/os2'
-archlibexp='/perl5/lib/5.10.0/os2'
+archlib='f:/lang/perl5/lib/5.10.0/os2'
+archlibexp='f:/lang/perl5/lib/5.10.0/os2'
 archname64=''
 archname='os2'
 archobjs='os2.o dl_os2.o'
@@ -110,8 +110,8 @@
 awk='awk'
 baserev='5.0'
 bash=''
-bin='/perl5/bin'
-binexp='/perl5/bin'
+bin='f:/lang/perl5/bin'
+binexp='f:/lang/perl5/bin'
 bison='bison'
 byacc='byacc'
 byteorder='1234'
@@ -136,9 +136,9 @@
 comm='comm'
 compress=''
 config_arg0='./configure'
-config_arg1='-Dprefix=/perl5'
+config_arg1='-Dprefix=f:/lang/perl5'
 config_argc='1'
-config_args='-Dprefix=/perl5'
+config_args='-Dprefix=f:/lang/perl5'
 contains='grep'
 cp='cp'
 cpio=''
@@ -770,25 +770,25 @@
 inc_version_list_init='0'
 incpath=''
 inews=''
-initialinstalllocation='/perl5/bin'
-installarchlib='/perl5/lib/5.10.0/os2'
-installbin='/perl5/bin'
+initialinstalllocation='f:/lang/perl5/bin'
+installarchlib='f:/lang/perl5/lib/5.10.0/os2'
+installbin='f:/lang/perl5/bin'
 installhtml1dir=''
 installhtml3dir=''
-installman1dir='/perl5/man/man1'
-installman3dir='/perl5/man/man3'
-installprefix='/perl5'
-installprefixexp='/perl5'
-installprivlib='/perl5/lib/5.10.0'
-installscript='/perl5/bin'
-installsitearch='/perl5/lib/site_perl/5.10.0/os2'
-installsitebin='/perl5/bin'
+installman1dir='f:/lang/perl5/man/man1'
+installman3dir='f:/lang/perl5/man/man3'
+installprefix='f:/lang/perl5'
+installprefixexp='f:/lang/perl5'
+installprivlib='f:/lang/perl5/lib/5.10.0'
+installscript='f:/lang/perl5/bin'
+installsitearch='f:/lang/perl5/lib/site_perl/5.10.0/os2'
+installsitebin='f:/lang/perl5/bin'
 installsitehtml1dir=''
 installsitehtml3dir=''
-installsitelib='/perl5/lib/site_perl/5.10.0'
-installsiteman1dir='/perl5/man/man1'
-installsiteman3dir='/perl5/man/man3'
-installsitescript='/perl5/bin'
+installsitelib='f:/lang/perl5/lib/site_perl/5.10.0'
+installsiteman1dir='f:/lang/perl5/man/man1'
+installsiteman3dir='f:/lang/perl5/man/man3'
+installsitescript='f:/lang/perl5/bin'
 installstyle='lib'
 installusrbinperl='define'
 installvendorarch=''
@@ -851,11 +851,11 @@
 mallocobj='malloc.o'
 mallocsrc='malloc.c'
 malloctype='void *'
-man1dir='/perl5/man/man1'
-man1direxp='/perl5/man/man1'
+man1dir='f:/lang/perl5/man/man1'
+man1direxp='f:/lang/perl5/man/man1'
 man1ext='1'
-man3dir='/perl5/man/man3'
-man3direxp='/perl5/man/man3'
+man3dir='f:/lang/perl5/man/man3'
+man3direxp='f:/lang/perl5/man/man3'
 man3ext='3'
 mips_type=''
 mistrustnm=''
@@ -908,17 +908,17 @@
 perl_patchlevel=''
 perladmin='paul@smedley.info'
 perllibs='-lsocket'
-perlpath='/perl5/bin/perl'
+perlpath='f:/lang/perl5/bin/perl'
 pg='pg'
 phostname='hostname'
 pidtype='pid_t'
 plibpth=''
 pmake=''
 pr=''
-prefix='/perl5'
-prefixexp='/perl5'
-privlib='/perl5/lib/5.10.0'
-privlibexp='/perl5/lib/5.10.0'
+prefix='f:/lang/perl5'
+prefixexp='f:/lang/perl5'
+privlib='f:/lang/perl5/lib/5.10.0'
+privlibexp='f:/lang/perl5/lib/5.10.0'
 procselfexe=''
 prototype='define'
 ptrsize='4'
@@ -952,8 +952,8 @@
 sPRIx64='"Lx"'
 sSCNfldbl='"Lf"'
 sched_yield='undef'
-scriptdir='/perl5/bin'
-scriptdirexp='/perl5/bin'
+scriptdir='f:/lang/perl5/bin'
+scriptdirexp='f:/lang/perl5/bin'
 sed='sed'
 seedfunc='srand48'
 selectminbits='1'
@@ -971,7 +971,7 @@
 sharpbang='#!'
 shmattype='void *'
 shortsize='2'
-shrpenv='env LD_RUN_PATH=/perl5/lib/5.10.0/os2/CORE'
+shrpenv='env LD_RUN_PATH=f:/lang/perl5/lib/5.10.0/os2/CORE'
 shsharp='true'
 sig_count='64'
 sig_name='ZERO HUP INT QUIT ILL TRAP ABRT EMT FPE KILL BUS SEGV SYS PIPE ALRM TERM URG STOP TSTP CONT CHLD TTIN TTOU IO XCPU XFSZ VTALRM PROF WINCH INFO USR1 USR2 BREAK RTMIN NUM34 NUM35 NUM36 NUM37 NUM38 NUM39 NUM40 NUM41 NUM42 NUM43 NUM44 NUM45 NUM46 NUM47 NUM48 NUM49 NUM50 NUM51 NUM52 NUM53 NUM54 NUM55 NUM56 NUM57 NUM58 NUM59 NUM60 NUM61 NUM62 RTMAX IOT CLD POLL '
@@ -980,25 +980,25 @@
 sig_num_init='0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 6, 20, 23, 0'
 sig_size='67'
 signal_t='void'
-sitearch='/perl5/lib/site_perl/5.10.0/os2'
-sitearchexp='/perl5/lib/site_perl/5.10.0/os2'
-sitebin='/perl5/bin'
-sitebinexp='/perl5/bin'
+sitearch='f:/lang/perl5/lib/site_perl/5.10.0/os2'
+sitearchexp='f:/lang/perl5/lib/site_perl/5.10.0/os2'
+sitebin='f:/lang/perl5/bin'
+sitebinexp='f:/lang/perl5/bin'
 sitehtml1dir=''
 sitehtml1direxp=''
 sitehtml3dir=''
 sitehtml3direxp=''
-sitelib='/perl5/lib/site_perl/5.10.0'
-sitelib_stem='/perl5/lib/site_perl'
-sitelibexp='/perl5/lib/site_perl/5.10.0'
-siteman1dir='/perl5/man/man1'
-siteman1direxp='/perl5/man/man1'
-siteman3dir='/perl5/man/man3'
-siteman3direxp='/perl5/man/man3'
-siteprefix='/perl5'
-siteprefixexp='/perl5'
-sitescript='/perl5/bin'
-sitescriptexp='/perl5/bin'
+sitelib='f:/lang/perl5/lib/site_perl/5.10.0'
+sitelib_stem='f:/lang/perl5/lib/site_perl'
+sitelibexp='f:/lang/perl5/lib/site_perl/5.10.0'
+siteman1dir='f:/lang/perl5/man/man1'
+siteman1direxp='f:/lang/perl5/man/man1'
+siteman3dir='f:/lang/perl5/man/man3'
+siteman3direxp='f:/lang/perl5/man/man3'
+siteprefix='f:/lang/perl5'
+siteprefixexp='f:/lang/perl5'
+sitescript='f:/lang/perl5/bin'
+sitescriptexp='f:/lang/perl5/bin'
 sizesize='4'
 sizetype='size_t'
 sleep=''
@@ -1014,7 +1014,7 @@
 srandom_r_proto='0'
 src='.'
 ssizetype='ssize_t'
-startperl='#!/perl5/bin/perl'
+startperl='#!f:/lang/perl5/bin/perl'
 startsh='#!/bin/sh.exe'
 static_ext=' '
 stdchar='char'
diff -uNr  lib/5.10.0/os2/CORE/config.h.org lib/5.10.0/os2/CORE/config.h
--- lib/5.10.0/os2/CORE/config.h.org	2007-12-27 17:21:08.000000000 +0900
+++ lib/5.10.0/os2/CORE/config.h	2011-11-17 19:32:50.000000000 +0900
@@ -894,8 +894,8 @@
  *	This symbol contains the ~name expanded version of ARCHLIB, to be used
  *	in programs that are not prepared to deal with ~ expansion at run-time.
  */
-#define ARCHLIB "/perl5/lib/5.10.0/os2"		/**/
-#define ARCHLIB_EXP "/perl5/lib/5.10.0/os2"		/**/
+#define ARCHLIB "f:/lang/perl5/lib/5.10.0/os2"     /**/
+#define ARCHLIB_EXP "f:/lang/perl5/lib/5.10.0/os2"     /**/
 
 /* ARCHNAME:
  *	This symbol holds a string representing the architecture name.
@@ -917,8 +917,8 @@
  *	This symbol, if defined, indicates that we'd like to relocate entries
  *	in @INC at run time based on the location of the perl binary.
  */
-#define BIN "/perl5/bin"	/**/
-#define BIN_EXP "/perl5/bin"	/**/
+#define BIN "f:/lang/perl5/bin"    /**/
+#define BIN_EXP "f:/lang/perl5/bin"    /**/
 #define PERL_RELOCATABLE_INC "undef" 		/**/
 
 /* CAT2:
@@ -2574,8 +2574,8 @@
  *	This symbol contains the ~name expanded version of PRIVLIB, to be used
  *	in programs that are not prepared to deal with ~ expansion at run-time.
  */
-#define PRIVLIB "/perl5/lib/5.10.0"		/**/
-#define PRIVLIB_EXP "/perl5/lib/5.10.0"		/**/
+#define PRIVLIB "f:/lang/perl5/lib/5.10.0"     /**/
+#define PRIVLIB_EXP "f:/lang/perl5/lib/5.10.0"     /**/
 
 /* CAN_PROTOTYPE:
  *	If defined, this macro indicates that the C compiler can handle
@@ -2667,8 +2667,8 @@
  *	This symbol contains the ~name expanded version of SITEARCH, to be used
  *	in programs that are not prepared to deal with ~ expansion at run-time.
  */
-#define SITEARCH "/perl5/lib/site_perl/5.10.0/os2"		/**/
-#define SITEARCH_EXP "/perl5/lib/site_perl/5.10.0/os2"		/**/
+#define SITEARCH "f:/lang/perl5/lib/site_perl/5.10.0/os2"      /**/
+#define SITEARCH_EXP "f:/lang/perl5/lib/site_perl/5.10.0/os2"      /**/
 
 /* SITELIB:
  *	This symbol contains the name of the private library for this package.
@@ -2690,9 +2690,9 @@
  *	removed.  The elements in inc_version_list (inc_version_list.U) can
  *	be tacked onto this variable to generate a list of directories to search.
  */
-#define SITELIB "/perl5/lib/site_perl/5.10.0"		/**/
-#define SITELIB_EXP "/perl5/lib/site_perl/5.10.0"		/**/
-#define SITELIB_STEM "/perl5/lib/site_perl"		/**/
+#define SITELIB "f:/lang/perl5/lib/site_perl/5.10.0"       /**/
+#define SITELIB_EXP "f:/lang/perl5/lib/site_perl/5.10.0"       /**/
+#define SITELIB_STEM "f:/lang/perl5/lib/site_perl"     /**/
 
 /* Size_t_size:
  *	This symbol holds the size of a Size_t in bytes.
@@ -4362,7 +4362,7 @@
  *	script to make sure (one hopes) that it runs with perl and not
  *	some shell.
  */
-#define STARTPERL "#!/perl5/bin/perl"		/**/
+#define STARTPERL "#!f:/lang/perl5/bin/perl"       /**/
 
 /* HAS_STDIO_STREAM_ARRAY:
  *	This symbol, if defined, tells that there is an array
